source/common/filter/config_discovery_impl.h

```diff
@@ -365,56 +438,86 @@ class FilterConfigProviderManagerImpl : public FilterConfigProviderManagerImplBa
     return std::make_unique<StaticFilterConfigProviderImpl<FactoryCb>>(config, filter_config_name);
   }
 
-  absl::string_view statPrefix() const override PURE;
-
-protected:
-  virtual ProtobufTypes::MessagePtr
-  getDefaultConfig(const ProtobufWkt::Any& proto_config, const std::string& filter_config_name,
-                   Server::Configuration::FactoryContext& factory_context,
-                   bool last_filter_in_filter_chain, const std::string& filter_chain_type,
-                   const absl::flat_hash_set<std::string>& require_type_urls) const PURE;
-
-private:
-  virtual std::unique_ptr<DynamicFilterConfigProviderImpl<FactoryCb>>
-  createFilterConfigProviderImpl(FilterConfigSubscriptionSharedPtr& subscription,
-                                 const absl::flat_hash_set<std::string>& require_type_urls,
-                                 Server::Configuration::FactoryContext& factory_context,
-                                 ProtobufTypes::MessagePtr&& default_config,
-                                 bool last_filter_in_filter_chain,
-                                 const std::string& filter_chain_type,
-                                 absl::string_view stat_prefix) PURE;
-};
-
-class HttpFilterConfigProviderManagerImpl
-    : public FilterConfigProviderManagerImpl<Http::FilterFactoryCb> {
-public:
   std::tuple<ProtobufTypes::MessagePtr, std::string>
   getMessage(const envoy::config::core::v3::TypedExtensionConfig& filter_config,
-             Server::Configuration::ServerFactoryContext& factory_context) const override;
+             Server::Configuration::ServerFactoryContext& factory_context) const override {
+    auto& factory = Config::Utility::getAndCheckFactory<Factory>(filter_config);
+    ProtobufTypes::MessagePtr message = Config::Utility::translateAnyToFactoryConfig(
+        filter_config.typed_config(),
+        factory_context.messageValidationContext().dynamicValidationVisitor(), factory);
+    return {std::move(message), factory.name()};
+  }
 
-  absl::string_view statPrefix() const override;
+  const absl::string_view& statPrefix() const override { return StatPrefix; }
```

Comment:
Should return `absl::string_view`

Reply:
ACK

Thanks for the comments. BTW, some background:
The change around constexpr absl::string_view to const absl::string_view&s is  to have the non-typed template parameter StatPrefix in this base class template parameter list, thus we can implement the method statPrefix in the base class here, and don't have to implement in the derived classes.  Also, only ref of string_view can be used as non-typed template parameter.   Now we backed out StatPrefix from the template parameter list, and implement the statPrefix() methods in the derived class, we don't need these changes any more.